lib/_http_client.js

```diff
@@ -22,6 +22,9 @@ function ClientRequest(options, cb) {
 
   if (typeof options === 'string') {
     options = url.parse(options);
+    if (!options.host && !options.hostname) {
```

Comment:
@ChALkeR Good point. I also think we need to check only the `hostname` here. I changed the check and included a test as well.

> Btw, how will the following code behave in such case (when the effective host includes the port, but not the domain)?

I tried `http.get('http://:80/')` and got the error `Error: getaddrinfo ENOTFOUND :80 :80:80`